src/Processors/Transforms/CachingTransform.cpp

```diff
@@ -0,0 +1,16 @@
+#include <Processors/Transforms/CachingTransform.h>
+
+namespace DB
+{
+
+void CachingTransform::transform(Chunk & chunk)
+{
+    if (!fits_into_memory)
+    {
+        return;
+    }
+
+    fits_into_memory = cache->insertChunk(cache_key, chunk.clone());
```

Comment:
Recalculate this `fits_into_memory`, using settings from Settings.h

Reply:
Need to be prepare if some exception will be thrown, lets that also be handled as part of API.